java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -505,8 +505,10 @@ abstract class AbstractArcHost(
      * Unregisters [Handle]s from [StorageProxy]s, and clears references to them from [Particle]s.
      */
     private suspend fun cleanupHandles(context: ParticleContext) {
-        withContext(context.particle.handles.dispatcher) {
-            context.particle.handles.reset()
+        if (!context.particle.handles.isEmpty()) {
```

Comment:
This fix doesn't explain why the handles field was empty tho. In the stack trace shown, the handles field should have been populated because your Manifest has a handle in it, therefore the archost should have created and installed one. Thus when cleanupHandles() was invoked, the 'data' handle should have been there.